plugins/statistics.koplugin/main.lua

```diff
@@ -2973,7 +2974,24 @@ function ReaderStatistics:getCurrentBookReadPages()
     return read_pages
 end
 
-function ReaderStatistics.onSync(local_path, cached_path, income_path)
+function ReaderStatistics:isSyncPossible()
+    return self.settings.sync_server ~= nil and self.settings.is_enabled
+end
+
+function ReaderStatistics:onSyncBookStats()
+    if not self:isSyncPossible() then return end
+
+    UIManager:show(InfoMessage:new {
+        text = _("Syncing book statistics, this might take a while…"),
+        timeout = 1,
+    })
```

Comment:
I guess the sync process is blocking, and this stays on until it's done, the timeout is there so it just disappear when the sync process is done?

Reply:
That's right, the same reason for using nextTick below